Snake/GameBoard.java

```diff
@@ -9,7 +19,7 @@
  * cannot move 180 degrees. Example: if the Snake is moving right, it cannot
  * immediately change its direction to left because it would run into itself.
  */
-class GameBoard  {
+class GameBoard extends JFrame {
```

Comment:
Is there a reason you made GameBoard extend JFrame?

Reply:
If I move the method to the Window class, all of the methods in GameBoard that call exit() need to have access to Window, but Window is currently private and causes errors if it's package.

Reply:
Pass the Window in the constructor for GameBoard so you can use it as the argument to JDIalog